iu_xray_data.py

```diff
@@ -0,0 +1,218 @@
+            if tag in self.pathologies]
+
+        label = torch.zeros(len(self.pathologies))
+        label[tag_index] = 1
+
+        # Get the findings
+        findings = items[1]
+        assert isinstance(findings, Findings)
+
+        if findings.content:
+            caption = findings.content
+        else:
+            caption = 'the lungs are normal.'
+
+        caption_token = list()
+        max_word_num = 0
+
+        for sentence in caption.split('. '):
+            sentence = sentence.replace('.', '').split()
+            if len(sentence) == 0 or len(sentence) == 1:
+                continue
+            tokens = self.vocab.map_tokens_to_ids_py(sentence).tolist()
+            tokens.append(self.vocab.eos_token_id)
+
+            max_word_num = max(max_word_num, len(tokens))
+            caption_token.append(tokens)
+        sentence_num = len(caption_token)
+
+        return image_tensor, label, caption_token, max_word_num, sentence_num
+
+    @staticmethod
+    def build_transform(tsfm_list):
+        t = []
+        for func, args in tsfm_list:
+            t.append(getattr(tfms, func)(**args))
+        return tfms.Compose(t)
+
+    def get_image(self, img_root, transforms):
+
+        # In this way, we can skip the ToPILImage in the data augmentations,
+        # speeding up the data loading
+        image = pil_loader(img_root)
+        image_tensor = transforms(image)
+        return image_tensor
+
+    @staticmethod
+    def default_hparams():
+        r"""Returns a dictionary of default hyperparameters.
+        See the specific subclasses for the details.
+        """
+        hparams = {
+            "img_root": None,
+            "text_root": None,
+            "vocab_path": None,
+            "transforms": None,
+            "pathologies": None,
+        }
+        return hparams
+
+
+class IU_XRay_Dataset(DatasetBase):
+    r"""Dataset for IU XRay
+    Args:
+        hparams (dict or HParams, optional): IU_XRay_Dataset hyperparameters.
+            Missing hyperparameters will be set to default values.
+            See :meth:`default_hparams` for the hyperparameter
+            structure and default values.
+                * datasource (Dict): hyperparameters for IU_XRay_DataSource
+        device: device to transer the data to
+    """
+    def __init__(self, hparams=None, device="cuda:0"):
+        self.source = IU_XRay_DataSource(hparams["datasource"])
+        super().__init__(self.source, hparams, device)
+
+    def collate(self, examples):
+        img_tensor, label, token_tensor, stop_prob = collate_fn(examples)
+
+        return tx.torch.data.Batch(
+            len(examples),
+            img_tensor=img_tensor,
+            label=label,
+            token_tensor=token_tensor,
+            stop_prob=stop_prob,
+        )
+
+    @staticmethod
+    def default_hparams():
+        r"""Returns a dictionary of default hyperparameters.
+        See the specific subclasses for the details.
```

Comment:
Can you fix the docstring and explain what the configurations are ?

Reply:
I thought I have included description of configurations at the class level. I changed the docstring here to be the same as the MLC. Let me know if something still need to be changed after I pushed the commit.